src/mctpd.c

```diff
@@ -2977,6 +3260,29 @@ static int emit_net_added(ctx *ctx, int net) {
 	return rc;
 }
 
+static int emit_interface_added(ctx *ctx, int ifindex) {
+	const char* ifname = NULL;
+	char *path = NULL;
+	int rc;
+
+	ifname = mctp_nl_if_byindex(ctx->nl, ifindex);
+	if (!ifname) {
+		warnx("BUG %s: no interface for ifindex %d", __func__, ifindex);
+		return -EPROTO;
+	}
+
+	path = interface_path(ifname);
+	if (path == NULL) {
+		warnx("%s: out of memory", __func__);
+		return -ENOMEM;
+	}
+	rc = sd_bus_emit_object_added(ctx->bus, dfree(path));
+	if (rc < 0)
+		warnx("%s: error emitting, %s", __func__, strerror(-rc));
```

Comment:
Just a tip on this: `warn` (rather than `warnx`) uses the errno value to automatically add the strerror value. So, if you prefer, you could do:

```c
if (rc < 0) {
    errno = -rc;
    warn("%s: error emitting", __func__);
}
```

\- and you would get the error string reported just the same.